library/Kubernetes/View/CronJobRenderer.php

```diff
@@ -1,82 +1,73 @@
 use Icinga\Module\Kubernetes\Common\Links;
 use ipl\Html\Attributes;
-use ipl\Html\BaseHtmlElement;
 use ipl\Html\HtmlDocument;
 use ipl\Html\HtmlElement;
 use ipl\Html\Text;
-use ipl\I18n\Translation;
 use ipl\Web\Widget\HorizontalKeyValue;
 use ipl\Web\Widget\Link;
-use ipl\Web\Widget\TimeAgo;
+use ipl\Web\Widget\StateBall;
 
-class CronJobListItem extends BaseListItem
+class CronJobRenderer extends BaseResourceRenderer
 {
-    use Translation;
+    public function assembleVisual($item, HtmlDocument $visual, string $layout): void
+    {
+        // TODO add icinga state then remove this function
+        $visual->addHtml(new StateBall('none', StateBall::SIZE_MEDIUM));
+    }
 
-    protected function assembleHeader(BaseHtmlElement $header): void
+    public function assembleCaption($item, HtmlDocument $caption, string $layout): void
     {
-        $header->addHtml(
-            $this->createTitle(),
-            new TimeAgo($this->item->created->getTimestamp())
-        );
+        // TODO add state reason then remove this function
+        $caption->addHtml(new Text('Placeholder for Icinga State Reason'));
     }
 
-    protected function assembleMain(BaseHtmlElement $main): void
+    public function assembleTitle($item, HtmlDocument $title, string $layout): void
     {
-        $main->addHtml(
-            $this->createHeader(),
-            $this->createFooter()
+        $title->addHtml(
+            new HtmlElement(
+                'span',
+                new Attributes(['class' => 'namespace-badge']),
+                new HtmlElement('i', new Attributes(['class' => 'icon kicon-namespace'])),
+                new Text($item->namespace)
+            ),
+            new Link(
+                (new HtmlDocument())->addHtml(
+                    new HtmlElement('i', new Attributes(['class' => 'icon kicon-cronjob'])),
+                    new Text($item->name)
+                ),
+                Links::cronjob($item),
+                new Attributes(['class' => 'subject'])
+            )
         );
     }
 
-    protected function assembleFooter(BaseHtmlElement $footer): void
+    public function assembleFooter($item, HtmlDocument $footer, string $layout): void
     {
-        if (isset($this->item->last_schedule_time)) {
-            $lastScheduleTime = $this->item->last_schedule_time->format('Y-m-d H:i:s');
+        if (isset($item->last_schedule_time)) {
+            $lastScheduleTime = $item->last_schedule_time->format('Y-m-d H:i:s');
         } else {
             $lastScheduleTime = $this->translate('None');
         }
 
-        if (isset($this->item->last_successful_time)) {
-            $lastSuccessfulTime = $this->item->last_successful_time->format('Y-m-d H:i:s');
+        if (isset($item->last_successful_time)) {
+            $lastSuccessfulTime = $item->last_successful_time->format('Y-m-d H:i:s');
         } else {
             $lastSuccessfulTime = $this->translate('None');
         }
```

Comment:
```suggestion
        $lastSuccessfulTime = $item-> last_successful_time?->format('Y-m-d H:i:s') ?? $this->translate('None');
```

Newline remaining from the two if blocks can be removed then.